cli/deno_dir.rs

```diff
@@ -4,6 +4,8 @@ use crate::cache::DiskCache;
 
 use std::path::PathBuf;
 
+pub use dirs::cache_dir;
```

Comment:
Very good point, it's gonna be quite hard to not incur a lot of overhead in this case - we parse the options and setup `DENO_DIR` when building the `ProcState`. Feels to me it might be much easier if we restructured how the update check is done and call it in various subcommands that create `ProcState`. It would be even easier if we checked for updates only in `deno run` subcommand.